[Bug] PAX: building error in Rocky8 because of unsuitable Protobuf version

### Apache Cloudberry version

main

### What happened

When trying to build the Cloudberry with enabling PAX, one error returned:

```
 $ make -j$(nproc) -C ~/cloudberry | tee ~/cloudberry/make-$(date "+%Y.%m.%d-%H.%M.%S").log
 
 
 -- Found Protobuf: /usr/lib64/libprotobuf.so;-lpthread (found version "3.5.0")
 CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
   Could NOT find Protobuf: Found unsuitable version "3.5.0", but required is
   at least "3.6.1" (found /usr/lib64/libprotobuf.so;-lpthread)
 Call Stack (most recent call first):
   /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:598 (_FPHSA_FAILURE_MESSAGE)
   /usr/share/cmake/Modules/FindProtobuf.cmake:650 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
   FindDependencies.cmake:3 (find_package)
   CMakeLists.txt:83 (include)


 -- Configuring incomplete, errors occurred!
 make[1]: *** [Makefile:52: build] Error 1
 make[1]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage'
 make: *** [GNUmakefile:32: all] Error 2
 make: Leaving directory '/home/gpadmin/cloudberry'
```

For the default protobuf version in Rockylinux 8:

```
[gpadmin@cdw ~]$ rpm -q  protobuf
protobuf-3.5.0-15.el8.x86_64
```

### What you think should happen instead

Can remove the protobuf version requirement or use the default protobuf version in Rocky Linux 8 as the minimal requirement.

### How to reproduce

```
./configure --prefix=/usr/local/cloudberry-db             --disable-external-fts             --enable-debug             --enable-cassert             --enable-debug-extensions             --enable-gpcloud             --enable-ic-proxy             --enable-mapreduce             --enable-orafce             --enable-orca             --enable-pxf            --enable-tap-tests             --with-gssapi             --with-ldap             --with-libxml             --with-lz4             --with-openssl             --with-pam             --with-perl             --with-pgport=5432            --with-python                         --with-ssl=openssl             --with-openssl         --with-uuid=e2fs             --with-includes=/usr/local/xerces-c/include            --with-pythonsrc-ext  --with-libraries=/usr/local/cloudberry-db/lib | tee ~/cloudberry/configure-$(date "+%Y.%m.%d-%H.%M.%S").log


make -j$(nproc) -C ~/cloudberry | tee ~/cloudberry/make-$(date "+%Y.%m.%d-%H.%M.%S").log
```

### Operating System

Rocky Linux 8.10 (Green Obsidian)

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

I conducted a test:

1. Set the protobuf from 3.6.1 to 3.5.0 for a minimal version requirement between Rocky 8 & 9 to avoid building protobuf separately:

```
[gpadmin@cdw cloudberry]$ git diff contrib/pax_storage/FindDependencies.cmake
diff --git a/contrib/pax_storage/FindDependencies.cmake b/contrib/pax_storage/FindDependencies.cmake
index 024e7ebf..cf9758cb 100644
--- a/contrib/pax_storage/FindDependencies.cmake
+++ b/contrib/pax_storage/FindDependencies.cmake
@@ -1,6 +1,6 @@
 ## protobuf
 include(FindProtobuf)
-find_package(Protobuf 3.6.1 REQUIRED)
+find_package(Protobuf 3.5.0 REQUIRED)
```

2. Then, I met another error while building `googlebench`:

```
CMake Error at src/cpp/contrib/googlebench/CMakeLists.txt:315 (message):
  Failed to determine the source files for the regular expression backend
```

I can build `googlebench` in a single directory outside of Cloudberry, but it fails when I try within Cloudberry. I'm not sure if I'm heading in the right direction, so I would appreciate any assistance.

Note:

```
-- Google Benchmark version: v1.9.2-3-g0da57b85, normalized to 1.9.2.3
-- Looking for shm_open in rt
-- Looking for shm_open in rt - found
-- Performing Test HAVE_CXX_FLAG_WALL
-- Performing Test HAVE_CXX_FLAG_WALL - Failed
-- Performing Test HAVE_CXX_FLAG_WEXTRA
-- Performing Test HAVE_CXX_FLAG_WEXTRA - Failed
-- Performing Test HAVE_CXX_FLAG_WSHADOW
-- Performing Test HAVE_CXX_FLAG_WSHADOW - Failed
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL - Failed
-- Performing Test HAVE_CXX_FLAG_WOLD_STYLE_CAST
-- Performing Test HAVE_CXX_FLAG_WOLD_STYLE_CAST - Failed
-- Performing Test HAVE_CXX_FLAG_WCONVERSION
-- Performing Test HAVE_CXX_FLAG_WCONVERSION - Failed
-- Performing Test HAVE_CXX_FLAG_WERROR
-- Performing Test HAVE_CXX_FLAG_WERROR - Failed
-- Performing Test HAVE_CXX_FLAG_PEDANTIC
-- Performing Test HAVE_CXX_FLAG_PEDANTIC - Failed
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS - Failed
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32 - Failed
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING - Failed
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS - Failed
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED - Failed
-- Performing Test HAVE_CXX_FLAG_WD654
-- Performing Test HAVE_CXX_FLAG_WD654 - Failed
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY - Failed
-- Performing Test HAVE_CXX_FLAG_COVERAGE
-- Performing Test HAVE_CXX_FLAG_COVERAGE - Failed
-- Compiling and running to test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX -- failed to compile
-- Compiling and running to test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
-- Compiling and running to test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- failed to compile
CMake Error at src/cpp/contrib/googlebench/CMakeLists.txt:315 (message):
  Failed to determine the source files for the regular expression backend
```

--- a/contrib/pax_storage/FindDependencies.cmake
+++ b/contrib/pax_storage/FindDependencies.cmake
@@ -1,6 +1,6 @@
 ## protobuf
 include(FindProtobuf)
-find_package(Protobuf 3.6.1 REQUIRED)
+find_package(Protobuf 3.5.0 REQUIRED)
 
 # ztsd
 # in our image snapshot, zstd is managed using pkg-config, so so the pkg-config method is used first here

--- a/contrib/pax_storage/src/cpp/storage/filter/pax_sparse_filter.h
+++ b/contrib/pax_storage/src/cpp/storage/filter/pax_sparse_filter.h
@@ -28,6 +28,7 @@
 #pragma once
 #include "comm/cbdb_api.h"
 
+#include <atomic>
 #include <map>
 #include <memory>
 #include <utility>
@@ -190,3 +191,4 @@ class PaxSparseFilter final {
 };  // class PaxSparseFilter
 
 }  // namespace pax
+

